src/main/duke/storage/Storage.java

```diff
@@ -0,0 +1,134 @@
+package duke.storage;
+
+import duke.data.TaskList;
+import duke.data.task.Deadline;
+import duke.data.task.Event;
+import duke.data.task.Task;
+import duke.data.task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * This class handles the loading and saving of the TaskList into the hard drive.
+ *
+ * @author Chesterwongz
+ */
+public class Storage {
+    private final String FILEPATH;
+    private static final ArrayList<Task> storageList = new ArrayList<>();
+    private static final String DATE_TIME_FORMAT = "dd-MM-yyyy HH:mm";
+
+    public Storage(String filePath) {
+        FILEPATH = filePath;
+    }
+
+    /**
+     * Reads and loads the taskList from txt file. Creates new file if it does not exist.
+     *
+     * @return An ArrayList of Tasks loaded from the txt file.
+     */
+    public ArrayList<Task> load() {
+        try {
+            File myObj = new File(FILEPATH);
+            Scanner myReader = new Scanner(myObj);
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                switch (data.charAt(0)) {
+                case 'T':
+                    Task toDoTask = new ToDo(data.substring(4));
+                    if (data.charAt(2) == 'X') {
+                        toDoTask.markAsDone();
+                    }
+                    storageList.add(toDoTask);
+                    break;
+                case 'D':
+                    readDeadline(data);
+                    break;
+                case 'E':
+                    readEvent(data);
+                    break;
+                default:
+                    // Do nothing and continue to scan the next line.
+                }
+            }
+            myReader.close();
+        } catch (FileNotFoundException e) {
+            createFile();
+        }
+        return storageList;
+    }
+
+    private void readEvent(String data) {
+        try {
+            String[] splitEvent = data.split("\\|", 4);
+            if (splitEvent.length == 4) {
+                DateTimeFormatter format = DateTimeFormatter.ofPattern(DATE_TIME_FORMAT);
+                LocalDateTime at = LocalDateTime.parse(splitEvent[3], format);
+                Task eventTask = new Event(splitEvent[2], at);
+                if (data.charAt(2) == 'X') {
+                    eventTask.markAsDone();
+                }
+                storageList.add(eventTask);
+            }
+        } catch (DateTimeParseException e) {
+            // Invalid data, do nothing.
```

Comment:
You might want to print a message indicating the correct/ideal format for date time instead of leaving it blank so that the user can use your program better. You can even include a reminder message regarding this upon startup!